server/internal/core/application/covenantless.go

```diff
@@ -778,10 +782,15 @@ func (s *covenantlessService) SignRoundTx(ctx context.Context, signedRoundTx str
 	defer s.currentRoundLock.Unlock()
 	currentRound := s.currentRound
 
-	combined, err := s.builder.VerifyAndCombinePartialTx(currentRound.UnsignedTx, signedRoundTx)
+	numSignedInputs, combined, err := s.builder.VerifyAndCombinePartialTx(
```

Comment:
Suggestion:
can we count the numSignedInputs from the psbt sent by the client BEFORE locking the current round mutex? This way we prevent a user spamming unsigned tx to "block" the others

Reply:
not sure it's possible with the given design of `VerifyAndCombinePartialTx` as it requires a source and destination tx